ivy/functional/frontends/torch/linalg.py

```diff
@@ -116,6 +116,38 @@ def svdvals(A, *, driver=None, out=None):
     return ivy.svdvals(A, out=out)
 
 
+@to_ivy_arrays_and_back
+@with_unsupported_dtypes({"1.11.0 and below": ("float16", "bfloat16")}, "torch")
+def tensorinv(input, ind=2, *, out=None):
+    not_invertible = "Reshaped tensor is not invertible"
+    prod_cond = "Tensor shape must satisfy prod(A.shape[:ind]) == prod(A.shape[ind:])"
+    input_shape = ivy.shape(input)
+    if ind > 0:
+        shape_ind_end = input_shape[:ind]
+        shape_ind_start = input_shape[ind:]
+        prod_ind_end = 1
+        prod_ind_start = 1
+        for i in shape_ind_start:
+            prod_ind_start *= i
+        for j in shape_ind_end:
+            prod_ind_end *= j
+        if prod_ind_end == prod_ind_start:
+            inverse_shape = shape_ind_start + shape_ind_end
+            input = ivy.reshape(input, shape=(prod_ind_end, -1))
+            inverse_shape_tuple = tuple([*inverse_shape])
+            if len(ivy.shape(input)) > 1 and (ivy.inv(input)).any():
+            # TODO: replace ivy.inv(input) with ivy.linalg.cond or torch.linalg.cond
+                inverse_tensor = ivy.inv(input)
+            else:
+                ret = ivy.reshape(input, shape=inverse_shape_tuple, out=out)
```

Comment:
I'm a little confused as to why you reshape here. Is this part of the c++ implementation?

Reply:
Yes, I tried to follow the same logic while implementing the function.

> ![clion64_9f7jPFEAqJ](https://user-images.githubusercontent.com/51514616/209119340-32326209-6ead-4ae1-b31e-c202cbeb527e.png) ![clion64_Tf8Q7y6hXd](https://user-images.githubusercontent.com/51514616/209119350-c4f4f392-c70c-44bb-9628-23317bf9000c.png) 

```C++
shape_ind_end.insert()
```

As far as I see, this part does that job. At the end of the function, returned result reshaped with this variable.